### ISSUE TITLE
关于图5a中的一点疑问

### ISSUE THREAD

**Issue by dawson-chen** [Zillwang/StepSearch#1]
我看到图5中 grpo-it 和 grpo-base都出现明显的崩溃现象。
<img width="664" alt="Image" src="https://github.com/user-attachments/assets/d5cb2596-7d1d-4a8f-b5cd-57c86c41c6fa" />

我也在做一个类似的工作，之前在做实验的时候也遇到了类似的现象，但主要是从工程角度去找原因。

> 这是我在verl仓库里提的issue  https://github.com/volcengine/verl/issues/1581#issuecomment-2906516142

但我看你们论文用改进的算法确实可以稳定训练，想请教一下你们是怎么解释这个崩溃现象的

**Comment by zxh20001117** [comment]
> Hi, • The crash is caused by the fact that the model output sequence contains some wrong symbol or logic, but still reward, the learning rate is too large, which leads to the deviation of gradient update, we guess GRPO N rollout may increase this kind of incorrect reward. • In our experiments with the 14B and 32B models, we observed enhanced performance in long-context understanding and instruction following, while model crashes became less frequent or occurred in later training steps.
> 
> For StepSearch The response length of the StepSearch method become shorter during the training process, while the response length of PPO and GRPO continue increase. In this case, it is easier for small models with limited long context ability to hallucinate and crash.
> 
> ![Image](https://github.com/user-attachments/assets/d3e9e2cd-6a34-47c6-9743-5a4aceac7e1c)
As one of the authors of this work, I found during my research that another cause of training crashes was the probabilistic mis-triggering of the format reward loophole or trick by the model, as documented in Table 8 in the paper's appendix.
![Image](https://github.com/user-attachments/assets/7f3a8a52-d186-49ba-8877-528f7830a3d5)
I also found that when using only SEARCH KEY SCORES acting on final tokens, or not using information redundancy penalties, the model was trained to a late stage that would occasionally produce unintended response formats (e.g. fabricated search documents, incorrect search tag pairs, replies to answer tag pairs followed by content, nested think tag pairs, and so on) could still be rewarded. This resulted in the model being unable to correct itself, leading to a training collapse.
The solution can be categorised into two parts:
1. Reduce the learning rate and increase the number of warm-up steps to make the learning rate smoother.
2. Directly modify the Verl code to discard the parameter update when there is an abnormal critic or reward value.
2) Add reasonable inhibition mechanisms (e.g. the information redundancy penalty proposed), or add intervening intermediate text (e.g. suggesting that the number of searches has reached the upper limit and that a reply to the 'answer' tag is now required) to the model's interaction behaviour with the outside world.
